### PR TITLE
DOC: Added ``CONTRIBUTING.rst``

### PR DESCRIPTION
**Pull request by Schefflera-Arboricola**
Added a contributor's guide containing some discussion points from talks with @seberg and @MarsBarLee at PyCon India. LMKWYT.

Thank you :)

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
I think adding the info on sprintable labels is worthwhile, but it probably fits best in the [existing contributing guide](https://github.com/numpy/numpy/blob/main/doc/source/dev/index.rst) instead of adding another document.

**Comment by charris** [comment]
I would like to add  tracking the PR and responding to comments and requests for changes. Making a PR is not the end of the proecess.

**Comment by seberg** [comment]
One thing with this was that `CONTRIBUTING.md`/`.rst` is a very typical enetry-point in many projects.  Defining one may be good for discoverability even if it would mostly link out to information elsewhere.

**Comment by rossbar** [comment]
> One thing with this was that CONTRIBUTING.md/.rst is a very typical enetry-point in many projects. 

Indeed - I think the main "feature" is that GitHub automatically links to it in a pop-up window whenever a first-time contributor opens an issue/PR (or whenever there's a change to the document). Whether or not that's a UX improvement is in the eye of the beholder :upside_down_face: .

I have no strong opinion either way - I find numpy's contributing guide quite discoverable since it's the landing page of the developer docs. OTOH in e.g. NetworkX we *do* use the `CONTRIBUTING.rst` and then just [`.. include::` it](https://github.com/networkx/networkx/blob/main/doc/developer/contribute.rst) to make sure the docs don't diverge. It's a little clunkier that way as any relative links don't work in the github-rendered page (they're still fine in the sphinx-generated docs), but YMMV!

**Comment by melissawm** [comment]
I think having a separate document with different content from the existing contributing guide is not a good idea, since these things very quickly get out of sync. Using includes or just adding links to the Guide in a CONTRIBUTING.md or rst would be useful, though!

**Comment by charris** [comment]
Close/reopen to check the macos failure.

File: doc/source/dev/index.rst [hunk 1]
```diff
@@ -19,6 +19,21 @@ we list them in alphabetical order):
...
+also NumPy is a pretty well-established project, so it's hard to
+make assumptions about an ideal "first-time-contributor". 
+So, that's why we don't mark issues with the "good-first-issue"
+label. Instead, you'll find [issues labeled "Sprintable."](https://github.com/numpy/numpy/labels/sprintable).
+These issues can either be:
+
+- **Easily fixed** when you have guidance from an experienced
+contributor (perfect for working in a sprint).
+- **A learning opportunity** for those ready to dive deeper,
+even if you're not in a sprint.
```
Comment (yuanx749) [inline doc/source/dev/index.rst:32]: I think indention here will fix the CI failure.
```suggestion
- **Easily fixed** when you have guidance from an experienced
  contributor (perfect for working in a sprint).
- **A learning opportunity** for those ready to dive deeper,
  even if you're not in a sprint. 
```

Comment (Schefflera-Arboricola) [reply doc/source/dev/index.rst]: Shouldn't the lint test catch this? That seems more logical to me.

File: doc/source/dev/index.rst [hunk 2]
```diff
@@ -19,6 +19,21 @@ we list them in alphabetical order):
...
+These issues can either be:
+
+- **Easily fixed** when you have guidance from an experienced
+  contributor (perfect for working in a sprint).
+- **A learning opportunity** for those ready to dive deeper,
+  even if you're not in a sprint. 
+
+Additionally, depending on your prior experience, some "Sprintable"
+issues might be easy, while others could be more challenging for you.
+
```
Comment (mattip) [inline doc/source/dev/index.rst:36]: This is RST not MD. I think this is what you wanted?

```suggestion
label. Instead, you'll find issues labeled `"Sprintable"`_.
These issues can either be:

- **Easily fixed** when you have guidance from an experienced
  contributor (perfect for working in a sprint).
- **A learning opportunity** for those ready to dive deeper,
  even if you're not in a sprint. 

Additionally, depending on your prior experience, some "Sprintable"
issues might be easy, while others could be more challenging for you.

_`"Sprintable": https://github.com/numpy/numpy/labels/sprintable
```

**Comment by mattip** [comment]
BLAS errors not related, see #27548. You would avoid these by adding `[skip actions][skip azp][skip cirrus]` to your commit message, see [here](https://numpy.org/doc/stable/dev/development_workflow.html#commands-to-skip-continuous-integration)

**Comment by mattip** [comment]
Thanks @Schefflera-Arboricola and @yuanx749

**Pull request merged**